### PR TITLE
add new kb for bitbucket-jenkins relation

### PR DESCRIPTION
**Pull request by sebasrevuelta**
# Thanks for improving Semgrep Docs 😀

### Please ensure

- [ ] A subject matter expert (SME) reviews the content
- [x] A technical writer reviews the content or PR
- [x] This change has no security implications or else you have pinged the security team
- [x] Redirects are added if the PR changes page URLs
- [x] If you have changed any header tag links (doc/#this-kind-of-anchor), update all instances of that link

### CODE REVIEW

File: docs/kb/semgrep-ci/bitbucket-triggering-events-to-jenkins.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+---
+tags:
+  - Bitbucket
+  - Jenkins
+  - webhooks
+description: How to trigger events from the Bitbucket server to a Jenkins instance
+---
+# Triggering events from Bitbucket to Jenkins
+
+## Motivation
```
Comment (s-santillan) [inline docs/kb/semgrep-ci/bitbucket-triggering-events-to-jenkins.md:10]:
```suggestion

```
Think it's fine to remove this :)

File: docs/kb/semgrep-ci/bitbucket-triggering-events-to-jenkins.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+A common scenario for many organizations is having:
```
Comment (s-santillan) [inline docs/kb/semgrep-ci/bitbucket-triggering-events-to-jenkins.md:12]:
```suggestion
This guide walks you through connecting your Bitbucket source code manager (SCM) with Jenkins, a CI/CD environment, through the use of **webhooks**. By connecting Bitbucket with Jenkins, you are able to customize jobs, such as your Semgrep job, based on push or pull events from Bitbucket.
```

- Trimmed some words and applied the [inverted pyramid](https://www.nngroup.com/articles/inverted-pyramid/).
- This method supports all Bitbucket plans, so there's no need to specify.

File: docs/kb/semgrep-ci/bitbucket-triggering-events-to-jenkins.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+* Bitbucket (Cloud or Server) as Source Code Management system to manage source code repositories.
```
Comment (s-santillan) [inline docs/kb/semgrep-ci/bitbucket-triggering-events-to-jenkins.md:13]:
```suggestion

```

File: docs/kb/semgrep-ci/bitbucket-triggering-events-to-jenkins.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+* Jenkins as CI/CD platform to run pipelines.
```
Comment (s-santillan) [inline docs/kb/semgrep-ci/bitbucket-triggering-events-to-jenkins.md:14]:
```suggestion

```

File: docs/kb/semgrep-ci/bitbucket-triggering-events-to-jenkins.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+So, it is crucial to have some kind of links between the two platforms.
```
Comment (s-santillan) [inline docs/kb/semgrep-ci/bitbucket-triggering-events-to-jenkins.md:15]:
```suggestion

```

File: docs/kb/semgrep-ci/bitbucket-triggering-events-to-jenkins.md [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+## Defining a Bitbucket webhook 
+One way to connect Bitbucket (Cloud or Server) to a Jenkins instance is defining a Bitbucket webhook.
+It will allow us to trigger builds on Bitbucket's push and pull requests events.
```
Comment (s-santillan) [inline docs/kb/semgrep-ci/bitbucket-triggering-events-to-jenkins.md:19]:
```suggestion
It allows you to trigger builds on Bitbucket's push and pull requests events.
```

File: docs/kb/semgrep-ci/bitbucket-triggering-events-to-jenkins.md [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+The steps to setup the webhook are the following:
+
+* For a specific Bitbucket repository click on **Repository Settings**->**Webhooks**->**Add Webhook**
+![image info](/img/kb/ci-bitbucket-defining-webhook.png)
+* Put a name to the webhook.
+* Specify the URL. It must be `https://<YOUR_JENKINS_SERVER>/bitbucket-hook/`
+* Select Triggers for “Push” and “Pull requests Created and Updated”.
+![image info](/img/kb/ci-bitbucket-select-triggers.png)
+
+Now the webhook is set on the Bitbucket side. You can use it from your Jenkins instance.
```
Comment (s-santillan) [inline docs/kb/semgrep-ci/bitbucket-triggering-events-to-jenkins.md:38]: How do users use it from the Jenkins instance?

File: docs/kb/semgrep-ci/bitbucket-triggering-events-to-jenkins.md [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+A common scenario for many organizations is having:
+* Bitbucket (Cloud or Server) as Source Code Management system to manage source code repositories.
+* Jenkins as CI/CD platform to run pipelines.
+So, it is crucial to have some kind of links between the two platforms.
+
+## Defining a Bitbucket webhook 
+One way to connect Bitbucket (Cloud or Server) to a Jenkins instance is defining a Bitbucket webhook.
+It will allow us to trigger builds on Bitbucket's push and pull requests events.
+
+There are two requirements to comply:
```
Comment (s-santillan) [inline docs/kb/semgrep-ci/bitbucket-triggering-events-to-jenkins.md:21]:
```suggestion
:::info Prerequisites
```
We can make a callout box.

File: docs/kb/semgrep-ci/bitbucket-triggering-events-to-jenkins.md [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+* Installing the plugin ["Bitbucket Push and Pull Request"](https://plugins.jenkins.io/bitbucket-push-and-pull-request/) on the Jenkins server.
```
Comment (s-santillan) [inline docs/kb/semgrep-ci/bitbucket-triggering-events-to-jenkins.md:22]:
```suggestion
* You must install the plugin [<i class="fas fa-external-link fa-xs"></i> Bitbucket Push and Pull Request](https://plugins.jenkins.io/bitbucket-push-and-pull-request/) on the Jenkins server.
```

We know have little icons that serve as signposts when users skim the docs! It looks like this on live:

File: docs/kb/semgrep-ci/bitbucket-triggering-events-to-jenkins.md [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+* Having network visibility between the two platforms.
```
Comment (s-santillan) [inline docs/kb/semgrep-ci/bitbucket-triggering-events-to-jenkins.md:23]:
```suggestion
* Network visibility between the two platforms.
:::
```
What does network visibility mean? Are there permissions, IP addresses, etc. that need to be configured?

File: docs/kb/semgrep-ci/bitbucket-triggering-events-to-jenkins.md [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+
+:::note
```
Comment (s-santillan) [inline docs/kb/semgrep-ci/bitbucket-triggering-events-to-jenkins.md:25]:
```suggestion

```

File: docs/kb/semgrep-ci/bitbucket-triggering-events-to-jenkins.md [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+Read the plugin documentation to review it is compatible with your Jenkins instance.
```
Comment (s-santillan) [inline docs/kb/semgrep-ci/bitbucket-triggering-events-to-jenkins.md:26]:
```suggestion

```

File: docs/kb/semgrep-ci/bitbucket-triggering-events-to-jenkins.md [hunk 13]
```diff
@@ -0,0 +1,38 @@
...
+:::
```
Comment (s-santillan) [inline docs/kb/semgrep-ci/bitbucket-triggering-events-to-jenkins.md:27]:
```suggestion

```

File: docs/kb/semgrep-ci/bitbucket-triggering-events-to-jenkins.md [hunk 14]
```diff
@@ -0,0 +1,38 @@
...
+
+The steps to setup the webhook are the following:
```
Comment (s-santillan) [inline docs/kb/semgrep-ci/bitbucket-triggering-events-to-jenkins.md:29]:
```suggestion
The steps to set up the webhook are the following:
```

File: docs/kb/semgrep-ci/bitbucket-triggering-events-to-jenkins.md [hunk 15]
```diff
@@ -0,0 +1,38 @@
...
+
+* For a specific Bitbucket repository click on **Repository Settings**->**Webhooks**->**Add Webhook**
```
Comment (s-santillan) [inline docs/kb/semgrep-ci/bitbucket-triggering-events-to-jenkins.md:31]:
```suggestion
* From your Bitbucket repository click on **Repository Settings > Webhooks > Add Webhook**
```

File: docs/kb/semgrep-ci/bitbucket-triggering-events-to-jenkins.md [hunk 16]
```diff
@@ -0,0 +1,38 @@
...
+![image info](/img/kb/ci-bitbucket-defining-webhook.png)
```
Comment (s-santillan) [inline docs/kb/semgrep-ci/bitbucket-triggering-events-to-jenkins.md:32]:
```suggestion
![A Bitbucket repository's Webhooks page](/img/kb/ci-bitbucket-defining-webhook.png)
```

File: docs/kb/semgrep-ci/bitbucket-triggering-events-to-jenkins.md [hunk 17]
```diff
@@ -0,0 +1,38 @@
...
+* Put a name to the webhook.
```
Comment (s-santillan) [inline docs/kb/semgrep-ci/bitbucket-triggering-events-to-jenkins.md:33]:
```suggestion
1. Enter a **Title** for your webhook.
```
(Following what I see in the screenshot.)

File: docs/kb/semgrep-ci/bitbucket-triggering-events-to-jenkins.md [hunk 18]
```diff
@@ -0,0 +1,38 @@
...
+* Specify the URL. It must be `https://<YOUR_JENKINS_SERVER>/bitbucket-hook/`
```
Comment (s-santillan) [inline docs/kb/semgrep-ci/bitbucket-triggering-events-to-jenkins.md:34]:
```suggestion
1. Specify the URL. It must follow the pattern: `https://<YOUR_JENKINS_SERVER>/bitbucket-hook/`
```

File: docs/kb/semgrep-ci/bitbucket-triggering-events-to-jenkins.md [hunk 19]
```diff
@@ -0,0 +1,38 @@
...
+* Select Triggers for “Push” and “Pull requests Created and Updated”.
```
Comment (s-santillan) [inline docs/kb/semgrep-ci/bitbucket-triggering-events-to-jenkins.md:35]:
```suggestion
1. Select the following **Triggers:** 
    1. In the **Repository** section, select **Push**.
    1. In the **Pull request** section, select **Created** and **Updated**.
```

File: docs/kb/semgrep-ci/bitbucket-triggering-events-to-jenkins.md [hunk 20]
```diff
@@ -0,0 +1,38 @@
...
+![image info](/img/kb/ci-bitbucket-select-triggers.png)
```
Comment (s-santillan) [inline docs/kb/semgrep-ci/bitbucket-triggering-events-to-jenkins.md:36]:
```suggestion
![Webhook set up page](/img/kb/ci-bitbucket-select-triggers.png)
```

**Review by s-santillan — CHANGES_REQUESTED** [review]
Thanks for this article, Sebas.

* Most of my notes are just suggestions you can easily commit.
* Can we specify something on **network visibility** and **how do we use the Bitbucket webhook on Jenkins's side**?

File: docs/kb/semgrep-ci/bitbucket-triggering-events-to-jenkins.md (hunk 7 above)
Comment (sebasrevuelta) [reply docs/kb/semgrep-ci/bitbucket-triggering-events-to-jenkins.md:34]: There will be a KB-second part and KB-third part explaining Jenkins side.

Comment (s-santillan) [reply docs/kb/semgrep-ci/bitbucket-triggering-events-to-jenkins.md:34]: Got it. Okay! I'm resolving this. :)

File: docs/kb/semgrep-ci/bitbucket-triggering-events-to-jenkins.md [hunk 21]
```diff
@@ -0,0 +1,34 @@
...
+
+This guide walks you through connecting your Bitbucket source code manager (SCM) with Jenkins, a CI/CD environment, through the use of **webhooks**. By connecting Bitbucket with Jenkins, you are able to customize jobs, such as your Semgrep job, based on push or pull events from Bitbucket.
+
+## Defining a Bitbucket webhook 
+One way to connect Bitbucket (Cloud or Server) to a Jenkins instance is defining a Bitbucket webhook.
+It allows you to trigger builds on Bitbucket's push and pull requests events.
+
+:::info Prerequisites
+* You must install the [<i class="fas fa-external-link fa-xs"></i> Bitbucket Push and Pull Request](https://plugins.jenkins.io/bitbucket-push-and-pull-request/) plugin on your the Jenkins server. This method requires that your Jenkins instance is compatible with this plugin.
+* Jenkins and Bitbucket must be visible to each other. It means Bitbucket will invoke Jenkins after a push or pull request event and Jenkins will send the result/comments to Bitbucket.
```
Comment (s-santillan) [inline docs/kb/semgrep-ci/bitbucket-triggering-events-to-jenkins.md:19]: In this line, you wrote:
>... Bitbucket will invoke Jenkins after a push or pull ...

The wording implies that this is the effect that happens AFTER the user sets up the webhook. So it's not actually something the user has to do beforehand? If so, can we remove the bullet?

**Review by s-santillan — APPROVED** [review]
@sebasrevuelta Approved with one comment on your reword - can you take a look?

File: docs/kb/semgrep-ci/bitbucket-triggering-events-to-jenkins.md (hunk 21 above)
Comment (sebasrevuelta) [reply docs/kb/semgrep-ci/bitbucket-triggering-events-to-jenkins.md:19]: Yes, we can remove the bullet.

**Pull request merged**